src/Symfony/Component/DependencyInjection/Alias.php

```diff
@@ -40,10 +41,42 @@ public function isPublic()
      * Sets if this Alias is public.
      *
      * @param bool $boolean If this Alias should be public
+     *
+     * @return $this
      */
     public function setPublic($boolean)
     {
         $this->public = (bool) $boolean;
+
+        return $this;
+    }
+
+    /**
+     * Sets if this Alias is private.
+     *
+     * When set, the "private" state has a higher precedence than "public".
+     * You can use it since version 3.4 to flag the "public" visibility
+     * as deprecated and turn the alias private in 4.0.
```

Comment:
This description looks weird to me, compared to the actual implementation. When reading it, I understand I should use `public=true, private=true` to make the public visibility deprecated.

However, the implementation actually allows to use `public=false, private=true` as well to make a service which stays available through the BC layer of accessing private services (so preventing inlining and removal).
Is this expected ?

Reply:
Correct, and the implemented behavior is the expected one to me yes.
So, the behavior is what you describe:
when "private" is set, we don't care anymore about what "public" is (and we shouldn't, that'd be too complex if we did, that's what I meant with "higher precedence".)
when "private" is set, the service is not inlined nor removed anymore (but that's internal details we shouldn't tell about), and as a consequence, the service is private but remains publicly accessible, with a deprecation notice along the way, which is exactly what we want.

Reply:
reworded. better?